### PR TITLE
feat(model): dataset model support

### PR DESCRIPTION
**Pull request by callmehiphop**
This adds Dataset Model support.

### CODE REVIEW

**Comment by callmehiphop** [comment]
@tswast @shollyman I think I have the basic implementation of the CRUD methods complete, but I'm not sure how to test it? I tried reviewing https://github.com/googleapis/google-cloud-python/pull/7562 to see how models are created there, but my python isn't so great that I could figure it out 😢. Any hints here would be greatly appreciated!

@sduskis @JustinBeckwith Should I include samples in this PR? Historically some one else has always written them, but it looks like I can pull the region tags from the PRs in other langs.

**Comment by tswast** [comment]
> Should I include samples in this PR? Historically some one else has always written them, but it looks like I can pull the region tags from the PRs in other langs.

Either you or @steffnay should write the samples. I think Steffany wanted to focus on Node.js samples this quarter.

**Comment by steffnay** [comment]
This is great. I can do the samples!

**Comment by steffnay** [comment]
This is great. I can do the samples!

File: src/model.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.delete((err, apiResponse) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
```
Comment (stephenplusplus) [inline src/model.ts]: Possibly the most nit-picky thing you've heard all year, but some of these sentences end in periods and others don't.

File: src/model.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * const metadata = {
+       *   friendlyName: 'thebestmodelever'
```
Comment (stephenplusplus) [inline src/model.ts]: Can it be "theBestModelEver", or is all lowercase really the deal?

File: src/model.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+ * const bigquery = new BigQuery();
+ * const dataset = bigquery.dataset('my-dataset');
+ *
+ * const model = dataset.model('my-model');
+ */
+class Model extends common.ServiceObject {
+  constructor(dataset: Dataset, id: string) {
+    const methods = {
+      /**
+       * Delete a model.
```
Comment (stephenplusplus) [inline src/model.ts]: Maybe "Delete this model." since we're in the context of a specific model.

File: src/model.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.get((err, model2, apiResponse) => {
```
Comment (stephenplusplus) [inline src/model.ts]: Can we just show it with the `err` argument, so we don't have to do the `model2` thing? The comment mentions how to get the metadata from the original object, so the natural question people would have is "what is the difference between model and model2? Do I need it?"

File: src/model.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+       * });
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [model2, apiResponse] = await model.get();
+       */
+      get: true,
+
+      /**
+       * Return the metadata associated with the Model.
```
Comment (stephenplusplus) [inline src/model.ts]: Sometimes it says "Model" and others "model." I guess it should always be "model" unless we're talking about a @google-cloud/bigquery "Model" object.

File: src/model.ts (hunk 1 above)
Comment (callmehiphop) [reply src/model.ts]: Truth be told I just copied a lot of the docs from existing docs, lol.

File: src/model.ts (hunk 2 above)
Comment (callmehiphop) [reply src/model.ts]: We can do that!

File: src/model.ts (hunk 3 above)
Comment (callmehiphop) [reply src/model.ts]: Going to follow suit with the other methods and use "Delete the model", implies the same thing I think.

File: src/model.ts (hunk 4 above)
Comment (callmehiphop) [reply src/model.ts]: Yeah, that example felt kinda weird, I just went that way for completeness.

File: src/model.ts (hunk 1 above)
Comment (stephenplusplus) [reply src/model.ts]: I figured!

File: src/model.ts (hunk 3 above)
Comment (stephenplusplus) [reply src/model.ts]: SGTM!

**Pull request merged**